src/rgw/rgw_op.cc

```diff
@@ -5284,6 +5284,62 @@ int RGWCopyObj::verify_permission(optional_yield y)
 					      RGW_PERM_READ)) { 
 	  return -EACCES;
 	}
+
+  // Check for tagging permissions
+  auto tagging_drctv = s->info.env->get("HTTP_X_AMZ_TAGGING_DIRECTIVE");
```

Comment:
rename tagging_drctv -> tagging_directive
rename itag_der to something more intuitive name

Reply:
Done